metadata-load/src/main/java/uk/ac/ebi/ampt2d/metadata/parser/AnalysisFileTypeFromSet.java

```diff
@@ -0,0 +1,157 @@
+            for (AnalysisFileType anaFile : analysisFile) {
+                subAnalysisFileList.add(anaFile);
+            }
+        }
+        return subAnalysisFileList;
+    }
+
+    public AnalysisSetType getAnalysisSet(String xmlStr) throws XmlException {
+        return ANALYSISSETDocument.Factory.parse(xmlStr).getANALYSISSET();
+    }
+
+    public List<AnalysisFileType> extractFromSqlXml(SQLXML sqlXml) {
+        String xmlStr;
+        List<AnalysisFileType> analysisFileList = new ArrayList<>();
+
+        try {
+            xmlStr = sqlXml.getString();
+            AnalysisSetType analysisSet;
+            try {
+                analysisSet = getAnalysisSet(xmlStr);
+                analysisFileList = extract(analysisSet);
+            } catch (XmlException e) {
+                logger.info("Unable to convert XML String to AnalysisSet: ", xmlStr);
+            }
+        } catch (SQLException e) {
+            logger.info("Unable to convert SQLXML Object to String: ", sqlXml.toString());
+        }
+        return analysisFileList;
+    }
+
+    private Properties loadPropertiesFile() throws IOException {
+        Properties properties = new Properties();
+        InputStream inFile = null;
+        try {
+            inFile = new FileInputStream(propertiesFile);
+            try {
+                properties.load(inFile);
+            } catch (IOException e) {
+                logger.info("Unable to load JDBC connection properties file");
+                throw e;
+            }
+        } catch (FileNotFoundException e) {
+            logger.info("JDBC connection properties file not found: " + propertiesFile);
+            throw e;
+        } finally {
+            if (inFile != null) {
+                inFile.close();
+            }
+        }
+        return properties;
+    }
+
+    private String createUrl(Properties appProperties) {
+        String url = appProperties.getProperty("OracleJDBC.url");
+        String host = appProperties.getProperty("OracleJDBC.host");
+        String protocol = appProperties.getProperty("OracleJDBC.protocol");
+        String serviceName = appProperties.getProperty("OracleJDBC.servicename");
+        String port = appProperties.getProperty("OracleJDBC.port");
+        url = url + ":@(DESCRIPTION =(ADDRESS_LIST =(ADDRESS = (PROTOCOL = " + protocol + ")(HOST = " + host + ")(PORT = " +
+                port + ")))(CONNECT_DATA =(SERVER = DEDICATED)(SERVICE_NAME = " + serviceName + ")))";
+
+        return url;
+    }
+
+    public List<AnalysisFileType> getAnalysisFileObject() throws IOException, SQLException {
+        String prepSql = "select ANALYSIS_XML from ERA.ANALYSIS";
+        Connection dbConnection = null;
+        PreparedStatement prepStatement = null;
+        ResultSet resultSet = null;
+        SQLXML sqlXml;
+        List<AnalysisFileType> analysisFileList = new ArrayList<>();
+
+        Properties appProperties = loadPropertiesFile();
```

Comment:
we can make use of seperate property bean .

Reply:
+1

Reply:
Done